docs/safe_lambda_deployments.rst

```diff
@@ -106,35 +106,55 @@ resource:
           PreTraffic: !Ref PreTrafficLambdaFunction
           PostTraffic: !Ref PostTrafficLambdaFunction
 
+  PreTrafficLambdaFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: preTrafficHook.handler
+      Policies:
+        - Version: "2012-10-17"
+          Statement:
+          - Effect: "Allow"
+            Action:
+              - "codedeploy:PutLifecycleEventHookExecutionStatus"
+            Resource: "*"
```

Comment:
Is this scope-able at all? We may need to use Conditions here to restrict this to the correct Resource.

Reply:
I don't think CodeDeploy supports any service-specific context keys. I've restricted the Resource from * to -->

arn:aws:codedeploy:[region]:[accountid]:deploymentgroup:*/*

Reply:
SAM/CF generates something like this
arn:aws:codedeploy:us-west-2:[accountId]:deploymentgroup:[stackname]-ServerlessDeploymentApplication-[uniqueId]/[stackName]-[lambdaFunctionName]DeploymentGroup-[uniqueId]

I don't think we should hardcode these things. What do you think?